infra/gcp/namespaces/ensure-namespaces.sh

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bash
+#
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script creates namespaces, applies roles and role bindings and
+# checks if assigned role binding is the only one in the namespace
+# for k8s-infra projects
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR=$(dirname "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")")
+# shellcheck source=../lib.sh
+. "${SCRIPT_DIR}/lib.sh"
+
+ROLE_FILENAME="namespace-user-role.yml"
+ROLE_BINDING_FILENAME="namespace-user-role-binding.yml"
+
+function usage() {
+    echo "usage: $0 <cluster> [--kubectl-path ...] [--kubeconfig-path ...]" >&2
+    echo "       --kubectl-path    | -p      path to kubectl command if not in \$PATH [optional]" >&2
+    echo "       --kubeconfig-path | -k      path to kubeconfig if not in: $HOME/.kube/config [optional]" >&2
+    echo "example: $0 my-cluster --kubectl-path /custom/path/kubectl --kubeconfig-path /custom/path/kubeconfig"
+    echo >&2
+}
+
+function parse_args() {
+  # positional args
+  args=()
+
+  # named args
+  while [ "$1" != "" ]; do
+      case "$1" in
+          -p | --kubectl-path )         KUBECTL="$2";            shift;;
+          -k | --kubeconfig-path )      KUBECONFIG_PATH="$2";    shift;;
+          * )                           args+=("$1")             # if no match, add it to the positional args
+      esac
+      shift # move to next kv pair
+  done
+
+  # restore positional args
+  set -- "${args[@]}"
+
+  # set defaults
+  if [[ -z "${KUBECONFIG_PATH}" ]]; then
+      KUBECONFIG_PATH="$HOME/.kube/config";
+  fi
+
+  if [[ -z "${KUBECTL}" ]]; then
+    if ! [ -x "$(command -v kubectl)" ]; then
+      echo >&2 "kubectl not found in \$PATH and --kubectl-path flag is not set. Aborting.";
+      exit 1;
+    else
+      KUBECTL="kubectl"
+    fi
+  fi
+
+  if [ "${#args[@]}" -lt 1 ]; then
+    usage
+    exit 1
+  fi
+
+  CLUSTER="${args[0]}"
+}
+
+function create_namespace() {
+  if [ $# != 1 ]; then
+    echo "create_namespace(name) requires 1 argument" >&2
+    return 1
+  fi
+  namespace="$1"
+
+  $KUBECTL create -c "$CLUSTER" --kubeconfig "$KUBECONFIG_PATH" namespace "$namespace"
```

Comment:
further pedanticness:  KUBECTL should be quoted, in case it includes a space

```
$ X="a b c"

$ $X
bash: a: command not found

$ "$X"
bash: a b c: command not found
```

Reply:
Fixed :-)